### ISSUE TITLE
"Length not known" error in GaussianMLPPolicy

### ISSUE THREAD

**Issue by antmaiorino** [openai/rllab#76]
I'm trying to experiment with rllab but I'm getting stuck when trying to execute the sample run provided [here](https://rllab.readthedocs.io/en/latest/user/experiments.html#running-experiments). When I initialize `GaussianMLPPolicy` I get an error I can't get around. I attached the error message I get from jupyter.

I'm on a Win10 enterprise machine and I'm installing everything in docker from the dockerfile provided on this github repo. The only thing I changed is that I set THEANO_FLAGS to use cpu instead of gpu.
After building the image I installed Theano and Lasagne followind the instructions on the [official lasagne docs](http://lasagne.readthedocs.io/en/latest/user/installation.html#stable-lasagne-release).
Any help would be appreciated.

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-32-7500884b62b7> in <module>()
      2     env_spec=env.spec,
      3     # The neural network policy should have two hidden layers, each with 32 hidden units.
----> 4     hidden_sizes=(32, 32)
      5 )

/srv/rllab-master/rllab/policies/gaussian_mlp_policy.py in __init__(self, env_spec, hidden_sizes, learn_std, init_std, adaptive_std, std_share_network, std_hidden_sizes, min_std, std_hidden_nonlinearity, hidden_nonlinearity, output_nonlinearity, mean_network, std_network, dist_cls)
     96         self.min_std = min_std
     97 
---> 98         mean_var, log_std_var = L.get_output([l_mean, l_log_std])
     99 
    100         if self.min_std is not None:

/opt/conda/envs/rllab3/lib/python3.5/site-packages/lasagne/layers/helper.py in get_output(layer_or_layers, inputs, **kwargs)
    183                                  "mapping this layer to an input expression."
    184                                  % layer)
--> 185             all_outputs[layer] = layer.get_output_for(layer_inputs, **kwargs)
    186     # return the output(s) of the requested layer(s) only
    187     try:

/srv/rllab-master/rllab/core/lasagne_layers.py in get_output_for(self, input, **kwargs)
     27         reshaped_param = TT.reshape(self.param, (1,) * (ndim - 1) + (self.num_units,))
     28         tile_arg = TT.concatenate([input.shape[:-1], [1]])
---> 29         tiled = TT.tile(reshaped_param, tile_arg, ndim=ndim)
     30         return tiled
     31 

/opt/conda/envs/rllab3/lib/python3.5/site-packages/theano/tensor/basic.py in tile(x, reps, ndim)
   4396                       (isinstance(r, TensorVariable) and
   4397                       r.dtype in ["int8", "int16", "int32", "int64"])
-> 4398                       for r in reps]):
   4399         raise ValueError("elements of reps must be scalars of integer dtype")
   4400     elif len(reps) != x.ndim:

/opt/conda/envs/rllab3/lib/python3.5/site-packages/theano/tensor/basic.py in <listcomp>(.0)
   4393     except TypeError:
   4394         raise ValueError("reps must be iterable")
-> 4395     if not numpy.all([isinstance(r, integer_types) or
   4396                       (isinstance(r, TensorVariable) and
   4397                       r.dtype in ["int8", "int16", "int32", "int64"])

/opt/conda/envs/rllab3/lib/python3.5/site-packages/theano/tensor/var.py in __iter__(self)
    512     def __iter__(self):
    513         try:
--> 514             for i in xrange(theano.tensor.basic.get_vector_length(self)):
    515                 yield self[i]
    516         except TypeError:

/opt/conda/envs/rllab3/lib/python3.5/site-packages/theano/tensor/basic.py in get_vector_length(v)
   3905             return stop - start
   3906 
-> 3907     raise ValueError("length not known")
   3908 
   3909 

ValueError: length not known
```

**Comment by dementrock** [comment]
Hi @antmaiorino, make sure you install Theano and Lasagne to the exact commit as used by rllab. You can find these commits in the environment.yml file.

**Comment by Beckschen** [comment]
Exactly I am also facing the same problem now. If you have already solved the problem, could you please write down the solution? Thank you ~

**Comment by antmaiorino** [comment]
Hi @Beckschen, I solved the problem following @dementrock advice. The point is that I was using a version of Theano and Lasagne that was newer than the one listed in the environment.yml file. Uninstalling them and reinstalling the correct version solved the problem for me.

**Comment by Beckschen** [comment]
Hi Antonio. Thanks. I tried this and finally solved the problem☺️




Best regards

Jieneng Chen
On 11/12/2017 09:18, Antonio wrote:

Hi @Beckschen, I solved the problem following @dementrock advice. The point is that I was using a version of Theano and Lasagne that was newer than the one listed in the environment.yml file. Uninstalling them and reinstalling the correct version solved the problem for me.
